panic: reflect.Value.Convert: value of type string cannot be converted to type *testing.T

The following program `sample.go` triggers a panic:

```go
package plugindemo

import (
	"testing"
)

func TestDemo(t *testing.T) {
	t.Errorf("got: %s, want: %s", "AAA", "AAA")
}
```

Expected result:
```console
$ go test .
// output
=== RUN   TestDemo
    demo_test.go:8: got: AAA, want: BBB
--- FAIL: TestDemo (0.00s)
FAIL
```

Got:
```console
$ yaegi test .
// output
panic: reflect.Value.Convert: value of type string cannot be converted to type *testing.T

goroutine 1 [running]:
reflect.Value.Convert(0xdcb8e0, 0xc000499430, 0x98, 0x109e520, 0xee2040, 0x0, 0x0, 0x0)
        /home/ldez/.gvm/gos/go1.15.2/src/reflect/value.go:2437 +0x229
github.com/traefik/yaegi/interp.convertLiteralValue(0xc00049be00, 0x109e520, 0xee2040)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/run.go:2840 +0xd3
github.com/traefik/yaegi/interp.callBin(0xc00049ba00)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/run.go:1116 +0x4ee
github.com/traefik/yaegi/interp.setExec.func1(0xc00049ba00)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/cfg.go:2337 +0xb6
github.com/traefik/yaegi/interp.setExec.func1(0xc00049bb00)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/cfg.go:2326 +0x262
github.com/traefik/yaegi/interp.setExec(0xc00049bb00)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/cfg.go:2340 +0x103
github.com/traefik/yaegi/interp.genRun.func1(0xc00049b000, 0xc0004af801)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/cfg.go:1893 +0x88
github.com/traefik/yaegi/interp.(*node).Walk(0xc00049b000, 0xc0004af838, 0x0)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/interp.go:211 +0xb5
github.com/traefik/yaegi/interp.(*node).Walk(0xc00049ad00, 0xc0004af838, 0x0)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/interp.go:215 +0x66
github.com/traefik/yaegi/interp.(*node).Walk(0xc00049a800, 0xc0004af838, 0x0)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/interp.go:215 +0x66
github.com/traefik/yaegi/interp.genRun(0xc00049a800, 0xc000150800, 0x0)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/cfg.go:1885 +0x65
github.com/traefik/yaegi/interp.(*Interpreter).importSrc(0xc0003ae000, 0xc00012e95a, 0x1d, 0x7fff1601ce4a, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/src.go:142 +0x92d
github.com/traefik/yaegi/interp.(*Interpreter).EvalTest(...)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/interp/interp.go:400
main.test(0xc00011a170, 0x1, 0x1, 0xf3fa20, 0xddd3a0)
        /home/ldez/sources/go/src/github.com/traefik/yaegi/cmd/yaegi/test.go:114 +0xe31
main.main()
        /home/ldez/sources/go/src/github.com/traefik/yaegi/cmd/yaegi/yaegi.go:121 +0x387
```

Same thing for `t.logf("%s %s", "A", "B")`

--- a/_test/m2/m2_test.go
+++ b/_test/m2/m2_test.go
@@ -1,0 +1,7 @@
+package m2
+
+import "testing"
+
+func TestM2(t *testing.T) {
+	t.Errorf("got %s, want %s", "AAA", "BBB")
+}

--- a/_test/variadic10.go
+++ b/_test/variadic10.go
@@ -1,0 +1,14 @@
+package main
+
+import (
+	"log"
+	"os"
+)
+
+func main() {
+	logger := log.New(os.Stdout, "test ", log.Lmsgprefix)
+	logger.Printf("args: %v %v", 1, "truc")
+}
+
+// Output:
+// test args: 1 truc

--- a/_test/variadic10.go
+++ b/_test/variadic10.go
@@ -8,7 +8,9 @@ import (
 func main() {
 	logger := log.New(os.Stdout, "test ", log.Lmsgprefix)
 	logger.Printf("args: %v %v", 1, "truc")
+	logger.Printf("args: %v %v %v", 1, "truc", 2)
 }
 
 // Output:
 // test args: 1 truc
+// test args: 1 truc 2

--- a/interp/run.go
+++ b/interp/run.go
@@ -1069,8 +1069,7 @@ func callBin(n *node) {
 	// A method signature obtained from reflect.Type includes receiver as 1st arg, except for interface types.
 	rcvrOffset := 0
 	if recv := n.child[0].recv; recv != nil && !isInterface(recv.node.typ) {
-		numIn, numChild := funcType.NumIn(), len(child)
-		if variadic > 0 && numIn > variadic || numIn > numChild {
+		if variadic > 0 || funcType.NumIn() > len(child) {
 			rcvrOffset = 1
 		}
 	}

--- a/interp/run.go
+++ b/interp/run.go
@@ -1039,14 +1039,6 @@ func call(n *node) {
 	}
 }
 
-// pindex returns definition parameter index for function call.
-func pindex(i, variadic int) int {
-	if variadic < 0 || i <= variadic {
-		return i
-	}
-	return variadic
-}
-
 func getFrame(f *frame, l int) *frame {
 	switch l {
 	case 0:
@@ -1077,7 +1069,8 @@ func callBin(n *node) {
 	// A method signature obtained from reflect.Type includes receiver as 1st arg, except for interface types.
 	rcvrOffset := 0
 	if recv := n.child[0].recv; recv != nil && !isInterface(recv.node.typ) {
-		if funcType.NumIn() > len(child) {
+		numIn, numChild := funcType.NumIn(), len(child)
+		if variadic > 0 && numIn >= numChild || numIn > numChild {
 			rcvrOffset = 1
 		}
 	}
@@ -1089,7 +1082,13 @@ func callBin(n *node) {
 	}
 
 	for i, c := range child {
-		defType := funcType.In(rcvrOffset + pindex(i, variadic))
+		var defType reflect.Type
+		if variadic >= 0 && i >= variadic {
+			defType = funcType.In(variadic)
+		} else {
+			defType = funcType.In(rcvrOffset + i)
+		}
+
 		switch {
 		case isBinCall(c):
 			// Handle nested function calls: pass returned values as arguments

--- a/interp/run.go
+++ b/interp/run.go
@@ -1070,7 +1070,7 @@ func callBin(n *node) {
 	rcvrOffset := 0
 	if recv := n.child[0].recv; recv != nil && !isInterface(recv.node.typ) {
 		numIn, numChild := funcType.NumIn(), len(child)
-		if variadic > 0 && numIn >= numChild || numIn > numChild {
+		if variadic > 0 && numIn > variadic || numIn > numChild {
 			rcvrOffset = 1
 		}
 	}

